src/table.py

```diff
@@ -387,9 +388,20 @@ def merge_chars(self, ordered_chars: list):
         x0, top, x1, bottom = objects_to_bbox(ordered_chars)
         doctop_adj = ordered_chars[0]["doctop"] - ordered_chars[0]["top"]
         upright = ordered_chars[0]["upright"]
-
         direction = 1 if (self.horizontal_ltr if upright else self.vertical_ttb) else -1
 
+        matrix = ordered_chars[0]["matrix"]
+
+        rotation = 0
+        if not upright and matrix[1] < 0:
+            ordered_chars = reversed(ordered_chars)
+            rotation = 270
+
+        if matrix[0] < 0 and matrix[3] < 0:
+            rotation = 180
+        elif matrix[1] > 0:
+            rotation = 90
+
```

Comment:
How come we don't check `upright` when setting rotation to 180 or 90?

And i wonder whether this code could be in the form `if ...: rotation=90 elif ...: rotation=180 elif ...: rotation=270`?

Reply:
The "upright" variable is sort of a reserved word in the original table extraction logic (pdfplumber). It always means 90° if set. I won't tear apart everything in the logic to make that more meaningful.
pdfplumber ignores any rotation angles other than 0 and 90. It is also incapable of correctly extracting multi-line content if rotation is not 0. But we now can do it.
As long as I am bound to use pdfplumber's own word extraction (and not PyMuPDF's more elegant and faster methods) I need to use other indicators for dealing with things outside pdfplumber's reach.